src/manifold/src/smoothing.cpp

```diff
@@ -501,6 +501,269 @@ class Partition {
 
 namespace manifold {
 
+// sharpenedEdges are referenced to the input Mesh, but the triangles have
+// been sorted in creating the Manifold, so the indices are converted using
+// meshRelation_.
+std::vector<Smoothness> Manifold::Impl::UpdateSharpenedEdges(
+    const std::vector<Smoothness>& sharpenedEdges) const {
+  std::unordered_map<int, int> oldHalfedge2New;
+  for (int tri = 0; tri < NumTri(); ++tri) {
+    int oldTri = meshRelation_.triRef[tri].tri;
+    for (int i : {0, 1, 2}) oldHalfedge2New[3 * oldTri + i] = 3 * tri + i;
```

Comment:
curious if a vector can be used instead, and maybe we can have parallelism here?

Reply:
I tried that first, but it's made difficult because there are potentially more triangles in the old indices than the new and I'm not sure how many.